### PR TITLE
Bump all plbase Python dependencies

### PR DESCRIPTION
**Pull request by nwalters512**
This is a rollup of all the Dependabot PRs for Python dependencies in the `prairielearn/plbase` image.

### CODE REVIEW

File: images/plbase/python-requirements.txt [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 pycryptodome==3.16.0
 pyquaternion==0.9.9
 regex==2022.10.31
-scipy==1.9.3
+scipy==1.10.0
 sympy==1.11.1
-rpy2==3.5.6
-pygraphviz==1.6
+rpy2==3.5.7
+pygraphviz==1.10
```
Comment (nwalters512) [inline images/plbase/python-requirements.txt:17]: Looks like something changed in `pygraphviz>=1.7` as I can no longer install any newer versions in the `plbase` image. It fails with the following error:

```
Failed to build pygraphviz
Installing collected packages: pygraphviz
  Running setup.py install for pygraphviz ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for pygraphviz did not run successfully.
  │ exit code: 1
  ╰─> [58 lines of output]
      running install
      /usr/local/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-310
      creating build/lib.linux-x86_64-cpython-310/pygraphviz
      copying pygraphviz/testing.py -> build/lib.linux-x86_64-cpython-310/pygraphviz
      copying pygraphviz/graphviz.py -> build/lib.linux-x86_64-cpython-310/pygraphviz
      copying pygraphviz/agraph.py -> build/lib.linux-x86_64-cpython-310/pygraphviz
      copying pygraphviz/scraper.py -> build/lib.linux-x86_64-cpython-310/pygraphviz
      copying pygraphviz/__init__.py -> build/lib.linux-x86_64-cpython-310/pygraphviz
      creating build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_close.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_drawing.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_node_attributes.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_subgraph.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_readwrite.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_string.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_graph.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_attribute_defaults.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_edge_attributes.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_scraper.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_unicode.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_layout.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_html.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/__init__.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      copying pygraphviz/tests/test_clear.py -> build/lib.linux-x86_64-cpython-310/pygraphviz/tests
      running egg_info
      writing pygraphviz.egg-info/PKG-INFO
      writing dependency_links to pygraphviz.egg-info/dependency_links.txt
      writing top-level names to pygraphviz.egg-info/top_level.txt
      reading manifest file 'pygraphviz.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.png' under directory 'doc'
      warning: no files found matching '*.txt' under directory 'doc'
      warning: no files found matching '*.css' under directory 'doc'
      warning: no previously-included files matching '*~' found anywhere in distribution
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '.svn' found anywhere in distribution
      no previously-included directories found matching 'doc/build'
      adding license file 'LICENSE'
      writing manifest file 'pygraphviz.egg-info/SOURCES.txt'
      copying pygraphviz/graphviz.i -> build/lib.linux-x86_64-cpython-310/pygraphviz
      copying pygraphviz/graphviz_wrap.c -> build/lib.linux-x86_64-cpython-310/pygraphviz
      running build_ext
      building 'pygraphviz._graphviz' extension
      creating build/temp.linux-x86_64-cpython-310
      creating build/temp.linux-x86_64-cpython-310/pygraphviz
      gcc -pthread -B /usr/local/compiler_compat -Wno-unused-result -Wsign-compare -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /usr/local/include -fPIC -O2 -isystem /usr/local/include -fPIC -I/usr/local/include/python3.10 -c pygraphviz/graphviz_wrap.c -o build/temp.linux-x86_64-cpython-310/pygraphviz/graphviz_wrap.o
      In file included from /usr/include/graphviz/gvc.h:17:0,
                       from pygraphviz/graphviz_wrap.c:2712:
      /usr/include/graphviz/types.h:49:10: fatal error: cgraph.h: No such file or directory
       #include <cgraph.h>
                ^~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> pygraphviz

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

Seems like we're missing some headers, or perhaps they're not being installed to the correct place?

Comment (nwalters512) [reply images/plbase/python-requirements.txt]: I filed an issue here: https://github.com/pygraphviz/pygraphviz/issues/447. It's not obvious to me if this is an issue with pygrpahviz itself, or with the way that the AL2 graphviz headers are set up.

**Comment by eliotwrobson** [comment]
My only comment is to use some tool like the following on the reqs file to make sure everything is ok: https://snyk.io/advisor/check/python

**Comment by nwalters512** [comment]
I ran `python3 -m pip check` locally and it didn't report any issues!

**Review by eliotwrobson — APPROVED** [review]

**Comment by jonatanschroeder** [comment]
I won't have time to test this PR properly, so I'll leave it to others to actually review. That said, do you think this should also include dependencies for the Python grader?

**Comment by nwalters512** [comment]
@jonatanschroeder I plan to follow up with the grader dependencies in another PR.

File: images/plbase/python-requirements.txt (hunk 1 above)
Comment (mwest1066) [reply images/plbase/python-requirements.txt]: Yeah, I've been fighting with this for a while. Thanks for filing an upstream report!

**Review by mwest1066 — APPROVED** [review]

**Pull request merged**